### PR TITLE
Refactor identityprovider. Add SiaIdentityProvider

### PR DESCRIPTION
**Pull request by tokle**

### CODE REVIEW

File: container-disc/src/main/java/com/yahoo/container/jdisc/athenz/impl/AthenzIdentityProviderImpl.java [hunk 1]
```diff
@@ -130,12 +113,12 @@ public String getService() {
     }
 
     @Override
-    public SSLContext getSslContext() {
+    public SSLContext getIdentitySslContext() {
         try {
             SSLContext sslContext = SSLContext.getInstance("TLSv1.2");
```
Comment (bjorncs) [inline container-disc/src/main/java/com/yahoo/container/jdisc/athenz/impl/AthenzIdentityProviderImpl.java]: Consider replacing this code with `AthenzSslContextBuilder` from _vespa-athenz_

File: container-disc/src/main/java/com/yahoo/container/jdisc/athenz/AthenzIdentityProvider.java [hunk 2]
```diff
@@ -7,8 +7,7 @@
  * @author mortent
  */
 public interface AthenzIdentityProvider {
```
Comment (bjorncs) [inline container-disc/src/main/java/com/yahoo/container/jdisc/athenz/AthenzIdentityProvider.java]: Consider moving this to `vespa-athenz`.

**Review by bjorncs — APPROVED** [review]
We'll have to rewrite our integration test before merging this.

**Comment by bjorncs** [comment]
`
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:testCompile (default-testCompile) on project container-disc: Compilation failure: Compilation failure:
[ERROR] /source/container-disc/src/test/java/com/yahoo/container/jdisc/athenz/impl/AthenzIdentityProviderImplTest.java:[87,47] cannot find symbol
[ERROR] symbol:   method getNToken()
[ERROR] location: variable identityProvider of type com.yahoo.container.jdisc.athenz.AthenzIdentityProvider
[ERROR] /source/container-disc/src/test/java/com/yahoo/container/jdisc/athenz/impl/AthenzIdentityProviderImplTest.java:[119,47] cannot find symbol
[ERROR] symbol:   method getNToken()
[ERROR] location: variable identityProvider of type com.yahoo.container.jdisc.athenz.AthenzIdentityProvider
[ERROR] /source/container-disc/src/test/java/com/yahoo/container/jdisc/athenz/impl/AthenzIdentityProviderImplTest.java:[158,47] cannot find symbol
[ERROR] symbol:   method getNToken()
[ERROR] location: variable identityProvider of type com.yahoo.container.jdisc.athenz.AthenzIdentityProvider
[ERROR] -> [Help 1]
[ERROR] `

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/package-info.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+@ExportPackage
```
Comment (tokle) [inline vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/package-info.java]: add copyright

File: athenz-identity-provider-service/src/main/java/com/yahoo/vespa/hosted/athenz/identityprovider/AthenzIdentityProviderImpl.java.orig [hunk 4]
```diff
@@ -0,0 +1,224 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (bjorncs) [inline athenz-identity-provider-service/src/main/java/com/yahoo/vespa/hosted/athenz/identityprovider/AthenzIdentityProviderImpl.java.orig]: `AthenzIdentityProviderImpl.java.orig` - Emacs accident?

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/package-info.java (hunk 3 above)
Comment (bjorncs) [reply vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/package-info.java]: Copyright still missing :)

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
+package com.yahoo.vespa.athenz.identity;
```
Comment (bjorncs) [inline vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java]: Add copyright

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import com.yahoo.vespa.athenz.api.AthenzService;
+import com.yahoo.vespa.athenz.tls.AthenzSslContextBuilder;
+
+import javax.net.ssl.SSLContext;
+import java.io.File;
+import java.nio.file.Paths;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+
+public class SiaIdentityProvider implements AthenzIdentityProvider {
```
Comment (bjorncs) [inline vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java]: Add author :)

File: container-disc/src/main/java/com/yahoo/container/jdisc/Ckms.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+package com.yahoo.container.jdisc;
+
+public interface Ckms {
```
Comment (bjorncs) [inline container-disc/src/main/java/com/yahoo/container/jdisc/Ckms.java]: Consider adding author

**Review by bjorncs — CHANGES_REQUESTED** [review]

File: athenz-identity-provider-service/src/main/java/com/yahoo/vespa/hosted/athenz/identityprovider/AthenzIdentityProviderImpl.java.orig (hunk 4 above)
Comment (tokle) [reply athenz-identity-provider-service/src/main/java/com/yahoo/vespa/hosted/athenz/identityprovider/AthenzIdentityProviderImpl.java.orig]: git reset accident ...

**Comment by tokle** [comment]
@bjorncs ptal

**Review by bjorncs — APPROVED** [review]

**Pull request merged**